sample/src/main/java/org/walletconnect/example/MainActivity.kt

```diff
@@ -1,11 +1,103 @@
 package org.walletconnect.example
 
-import androidx.appcompat.app.AppCompatActivity
+import android.graphics.drawable.ColorDrawable
 import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import org.walletconnect.example.databinding.ActivityMainBinding
+import org.walletconnect.example.extension.getCurrentDestination
+import org.walletconnect.example.extension.getNavController
+import org.walletconnect.example.wallet.WalletViewModel
+import org.walletconnect.example.wallet.ui.SessionProposalDialog
+import org.walletconnect.example.wallet.ui.ShowSessionProposalDialog
+import org.walletconnect.example.wallet.ui.ToggleBottomNav
 
 class MainActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+    private val viewModel: WalletViewModel by viewModels()
+    private var sessionProposalDialog: SessionProposalDialog? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        supportActionBar?.setBackgroundDrawable(
+            ColorDrawable(
+                ContextCompat.getColor(
+                    this,
+                    R.color.blue
+                )
+            )
+        )
+
+        viewModel.eventFlow.observe(this, { event ->
+            when (event) {
+                is ToggleBottomNav -> {
+                    if (event.shouldShown) {
+                        binding.bottomNav.visibility = View.VISIBLE
+                    } else {
+                        binding.bottomNav.visibility = View.GONE
+                    }
+                }
+                is ShowSessionProposalDialog -> {
+                    sessionProposalDialog = SessionProposalDialog(this, {
+                        viewModel.approve()
+                    }, {
+                        viewModel.reject()
+                    }).apply {
+                        setContent(event.proposal)
+                        show()
+                    }
+                }
+            }
+        })
+        setBottomNavigation()
+    }
+
+    private fun setBottomNavigation() {
+        binding.bottomNav.setOnNavigationItemSelectedListener { item ->
+            when (item.itemId) {
+                R.id.wallet -> if (getNavController().currentDestination?.id != R.id.walletFragment)
+                    getNavController().navigate(R.id.action_dappFragment_to_walletFragment)
+                R.id.dapp -> if (getNavController().currentDestination?.id != R.id.dappFragment)
+                    getNavController().navigate(R.id.action_walletFragment_to_dappFragment)
+            }
+            true
+        }
```

Comment:
This can be replaced by 
```suggestion
NavigationUI.setupWithNavController(binding.bottomNav, navHostFragment.findNavController())
```